### PR TITLE
Disable sampling in preprocessing or when it results in too few rows

### PR DESCRIPTION
**Pull request by ShreyaR**

### CODE REVIEW

File: ludwig/data/preprocessing.py [hunk 1]
```diff
@@ -1119,8 +1119,14 @@ def build_dataset(
 
     sample_ratio = global_preprocessing_parameters["sample_ratio"]
     if sample_ratio < 1.0:
-        logger.debug(f"sample {sample_ratio} of data")
-        dataset_df = dataset_df.sample(frac=sample_ratio)
+        if len(dataset_df) * sample_ratio < 1:
+            warnings.warn(
+                f"sample_ratio {sample_ratio} is too small for dataset of length {len(dataset_df)}. "
+                f"Please increase sample_ratio or use a larger dataset."
+            )
```
Comment (arnavgarg1) [inline ludwig/data/preprocessing.py:1126]: Should we raise an error instead?

**Review by jppgks — APPROVED** [review]

File: ludwig/data/preprocessing.py (hunk 1 above)
Comment (ShreyaR) [reply ludwig/data/preprocessing.py]: done

**Comment by arnavgarg1** [comment]
@ShreyaR can we add a test?

File: ludwig/data/preprocessing.py [hunk 2]
```diff
@@ -1086,13 +1089,30 @@ def build_dataset(
...
+        mode: One of ['training', 'prediction']
+        metadata: Training set metadata if available
+        backend: Backend
+        random_seed: Random seed
+        skip_save_processed_input: Whether to skip saving the processed input
+        callbacks: List of callbacks
+
+    Returns:
+        A tuple of (dataset, metadata)
+    """
```
Comment (arnavgarg1) [inline ludwig/data/preprocessing.py:1114]: 🙇‍♂️ thanks for adding this docstring

File: tests/integration_tests/test_preprocessing.py [hunk 3]
```diff
@@ -63,6 +64,18 @@ def test_sample_ratio(backend, tmpdir, ray_cluster_2cpu):
...
 
+    # Check that sample ratio is disabled when doing preprocessing for prediction
+    dataset, _ = preprocess_for_prediction(
+        model.config_obj.to_dict(),
+        dataset=data_csv,
+        training_set_metadata=training_set_metadata,
+        split=FULL,
+        include_outputs=True,
+        backend=model.backend,
+    )
```
Comment (arnavgarg1) [inline tests/integration_tests/test_preprocessing.py:75]: don't we need to pass in `mode=prediction`?

Comment (arnavgarg1) [reply tests/integration_tests/test_preprocessing.py:75]: nvm, it's an implementation detail got it :)

**Review by arnavgarg1 — APPROVED** [review]
Nice work! 🚢

**Pull request merged**